### PR TITLE
chore: update flakes, melange

### PR DESCRIPTION
**Pull request by anmonteiro**
- run `nix flake update` to get the newest melange hash
  - tests dune with the most recent melange, using dune to compile its own stdlib / runtime

### CODE REVIEW

File: test/blackbox-tests/test-cases/melange/merlin.t [hunk 1]
```diff
@@ -3,7 +3,10 @@
   $ ocamlc_where="$(ocamlc -where)"
   $ export BUILD_PATH_PREFIX_MAP="/OCAMLC_WHERE=$ocamlc_where:$BUILD_PATH_PREFIX_MAP"
   $ melc_where="$(melc -where)"
-  $ export BUILD_PATH_PREFIX_MAP="/MELC_WHERE=$melc_where:$BUILD_PATH_PREFIX_MAP"
+  $ IFS=: read -r melc_stdlib melc_stdlib2 melc_stdlib3 <<< "$melc_where"
+  $ export BUILD_PATH_PREFIX_MAP="/MELC_WHERE1=$melc_stdlib:$BUILD_PATH_PREFIX_MAP"
+  $ export BUILD_PATH_PREFIX_MAP="/MELC_WHERE2=$melc_stdlib2:$BUILD_PATH_PREFIX_MAP"
+  $ export BUILD_PATH_PREFIX_MAP="/MELC_WHERE3=$melc_stdlib3:$BUILD_PATH_PREFIX_MAP"
```
Comment (anmonteiro) [inline test/blackbox-tests/test-cases/melange/merlin.t:9]: not happy about this at all. Here's the problem:

- `BUILD_PATH_PREFIX` is `:` separated
- the result of `melange --where` is `:` separated

Happy to hear alternatives. In the meantime this makes the tests pass.

Comment (rgrinberg) [reply test/blackbox-tests/test-cases/melange/merlin.t:9]: Looks fine to me. I don't see  the problem apart from hard coding it to use 3 paths. I would also substitute them all to `MELC_STDLIB`.

**Comment by rgrinberg** [comment]
@Alizter does this flake update work for you?

**Comment by Alizter** [comment]
Nope:
```
error: builder for '/nix/store/8pj0y0m1b8225cc5xisyxcbdwr6bb89h-ocaml4.14.1-melange-dev.drv' failed with exit code 1;
       last 10 log lines:
       > no configure script, doing nothing
       > building
       > File "dune-project", line 1, characters 11-14:
       > 1 | (lang dune 3.7)
       >                ^^^
       > Error: Version 3.7 of the dune language is not supported.
       > Supported versions of this extension in version 3.7 of the dune language:
       > - 1.0 to 1.12
       > - 2.0 to 2.9
       > - 3.0 to 3.6
       For full logs, run 'nix log /nix/store/8pj0y0m1b8225cc5xisyxcbdwr6bb89h-ocaml4.14.1-melange-dev.drv'.
error: 1 dependencies of derivation '/nix/store/4yf6f1cvk6m9z8x93hnx1k3nrkh6g1b6-nix-shell-env.drv' failed to build

```

**Comment by anmonteiro** [comment]
@Alizter which command did you run? I might have to add an override somewhere.

**Comment by Alizter** [comment]
@anmonteiro I just did `nix develop`

**Comment by Alizter** [comment]
Works now.

**Review by Alizter — APPROVED** [review]

**Comment by anmonteiro** [comment]
oh? I'm not sure what I did to fix it.

**Pull request merged**